zipline/sources/test_source.py

```diff
@@ -88,7 +87,7 @@ def advance_current(cur):
 
     # yield count trade events, all on trading days, and
     # during trading hours.
-    for i in range(count):
+    while cur < end:
```

Comment:
I think that might help reduce some cases where the gen is yielding nothing and causing something to silently not run.